src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -95,58 +95,20 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
     }
 
     public async deleteTreeItemImpl(): Promise<void> {
-        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
-        await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
-    }
-
-    public async getClientConfig(): Promise<ClientConfig> {
-        const { username, password } = await this.parent.getCredentials();
-
-        if (username && password) {
-            const ssl: ConnectionOptions = {
-                // Always provide the certificate since it is accepted even when SSL is disabled
-                // Certificate source: https://aka.ms/AA7wnvl
-                ca: BaltimoreCyberTrustRoot
-            };
-
-            const host: string = nonNullProp(this.parent.server, 'fullyQualifiedDomainName');
-            const clientConfig: ClientConfig = { user: username, password, ssl, host, port: 5432, database: this.databaseName };
-
-            // Ensure the client config is valid before returning
-            const client: Client = new Client(clientConfig);
+        if (this.parent.server) {
+            const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+            await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
+        } else {
+            const username_connString = this.parent.connectionString.username;
+            const password_connString = this.parent.connectionString.password;
+            const host = this.parent.connectionString.hostName;
+            const config = { user: username_connString, password: password_connString, host, port: 5432 };
             try {
-                await client.connect();
-                return clientConfig;
-            } finally {
-                await client.end();
+                await dropdb(config, this.databaseName);
+            } catch (error) {
+                throw error;
```

Comment:
Haha just remove the try/catch if it's just gonna throw the error that would've been thrown anyways